model/adjuster/span_id_deduper.go

```diff
@@ -55,12 +55,14 @@ type spanIDDeduper struct {
 // groupSpansByID groups spans with the same ID returning a map id -> []Span
 func (d *spanIDDeduper) groupSpansByID() {
 	spansByID := make(map[model.SpanID][]*model.Span)
-	for _, span := range d.trace.Spans {
-		if spans, ok := spansByID[span.SpanID]; ok {
-			// TODO maybe return an error if more than 2 spans found
-			spansByID[span.SpanID] = append(spans, span)
-		} else {
-			spansByID[span.SpanID] = []*model.Span{span}
+	if d.trace.Spans != nil {
```

Comment:
Is this really the cause? I thought nil could be used just fine with range. It would be easy to prove with a unit test ;-)

Reply:
Yes, range over nil slice is fine. What problem is this trying to solve?

Reply:
I will be checking it with unit tests